test/spec/modules/permutiveRtdProvider_spec.js

```diff
@@ -29,6 +31,128 @@ describe('permutiveRtdProvider', function () {
+      return () => removeLocalStorage(dataToStore)
+    }
+
+    const setWindowPermutivePrebid = (getPermutiveRtdConfig) => {
+      // Read from Permutive
+      const backup = window.permutive
+
+      deepSetValue(window, 'permutive.addons.prebid', {
+        getPermutiveRtdConfig,
+      })
+
+      // Cleanup
+      return () => window.permutive = backup
+    }
+
+    it('should return default values', function () {
+      const config = getModuleConfig({})
+      expect(config).to.deep.equal(getDefaultConfig())
+    })
+
+    it('should override deeply on custom config', function () {
+      const defaultConfig = getDefaultConfig()
+
+      const customModuleConfig = { waitForIt: true, params: { acBidders: ['123'] } }
+      const config = getModuleConfig(customModuleConfig)
+
+      expect(config).to.deep.equal(mergeDeep(defaultConfig, customModuleConfig))
+    })
+
+    it('should override deeply on cached config', function () {
+      const defaultConfig = getDefaultConfig()
+
+      const cachedParamsConfig = { acBidders: ['123'] }
+      const cleanupCache = storeConfigInCacheAndInit(cachedParamsConfig)
+
+      const config = getModuleConfig({})
+
+      expect(config).to.deep.equal(mergeDeep(defaultConfig, liftToParams(cachedParamsConfig)))
+
+      // Cleanup
+      cleanupCache()
+    })
+
+    it('should override deeply on Permutive Rtd config', function () {
+      const defaultConfig = getDefaultConfig()
+
+      const permutiveRtdConfigParams = { acBidders: ['123'], overwrites: { '123': true } }
+      const cleanupPermutive = setWindowPermutivePrebid(function () {
+        return permutiveRtdConfigParams
+      })
+
+      const config = getModuleConfig({})
+
+      expect(config).to.deep.equal(mergeDeep(defaultConfig, liftToParams(permutiveRtdConfigParams)))
+
+      // Cleanup
+      cleanupPermutive()
+    })
+
+    it('should handle calling Permutive method throwing error', function () {
+      const defaultConfig = getDefaultConfig()
+
+      const cleanupPermutive = setWindowPermutivePrebid(function () {
+        throw new Error()
+      })
+
+      const config = getModuleConfig({})
+
+      expect(config).to.deep.equal(defaultConfig)
+
+      // Cleanup
+      cleanupPermutive()
+    })
+
+    it('should override deeply in priority order', function () {
+      const defaultConfig = getDefaultConfig()
+
+      const cachedConfig = { acBidders: ['123'] }
+      const cleanupCache = storeConfigInCacheAndInit(cachedConfig)
+
+      // Read from Permutive
+      const permutiveRtdConfig = { acBidders: ['456'] }
+      const cleanupPermutive = setWindowPermutivePrebid(function () {
+        return permutiveRtdConfig
+      })
+
+      const customModuleConfig = { params: { acBidders: ['789'], maxSegs: 499 } }
+      const config = getModuleConfig(customModuleConfig)
+
+      const configMergedInPriorityOrder = mergeDeep(defaultConfig, liftToParams(cachedConfig), liftToParams(permutiveRtdConfig), customModuleConfig)
```

Comment:
you want the configuration coming from window.permutive to override how the publisher configures the module in prebid?

Reply:
The name is misleading here, the configs are in reverse priority order as configs are merged left to right. So the priority is,

1. `customModuleConfig` <- set by publisher with `pbjs.setConfig`
2. `permutiveRtdConfig` <- set by config publisher configures inside Permutive.
3. `cacheConfig` <- Params stored in LS
4. `defaultConfig`

Therefore, anything set by the publisher will take priority over what is set within Permutive. I'll add some comments explaining this.

Reply:
thanks!